File: webforj-components/webforj-html-elements/src/test/java/com/webforj/component/html/elements/ImgTest.java
package com.webforj.component.html.elements;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertEquals;

import com.webforj.component.element.PropertyDescriptorTester;
import org.junit.jupiter.api.Test;

class ImgTest {

  @Test
  void shouldConstructWithSrcAndAlt() {
    String src = "https://example.com";
    String alt = "Example";

    Img img = new Img(src, alt);
    assertEquals(src, img.getSrc());
    assertEquals(alt, img.getAlt());
  }

  @Test
  void shouldConstructWithSrc() {
    String src = "https://example.com";

    Img img = new Img(src);
    assertEquals(src, img.getSrc());
  }

  @Test
  void shouldSetGetProperties() {
    Img component = new Img();

    assertDoesNotThrow(() -> {
      PropertyDescriptorTester.run(Img.class, component);
    });
  }
}

File: webforj-kotlin/src/test/kotlin/com/webforj/kotlin/dsl/component/html/elements/ImgTest.kt
package com.webforj.kotlin.dsl.component.html.elements

import com.webforj.component.html.elements.Div
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class ImgTest {
    lateinit var root: Div

    @BeforeEach
    fun setUp() {
        root = Div()
    }

    @Test
    @DisplayName("Create Img with source")
    fun shouldCreateImgWithSrc() {
        val src = "src"
        val img = root.img(src)
        assertEquals(src, img.src)
        assertEquals("", img.alt)
        assertTrue(root.hasComponent(img))
    }

    @Test
    @DisplayName("Create Img with alt text")
    fun shouldCreateImgWithAlt() {
        val alt = "alt"
        val img = root.img(alt = alt)
        assertEquals("", img.src)
        assertEquals(alt, img.alt)
        assertTrue(root.hasComponent(img))
    }

    @Test
    @DisplayName("Create Img with source and alt text")
    fun shouldCreateImgWithSrcAndAlt() {
        val src = "src"
        val alt = "alt"
        val img = root.img(src, alt)
        assertEquals(src, img.src)
        assertEquals(alt, img.alt)
        assertTrue(root.hasComponent(img))
    }

}
